aws_lambda_powertools/utilities/data_classes/shared_functions.py

```diff
@@ -21,12 +21,30 @@ def base64_decode(value: str) -> str:
     return base64.b64decode(value).decode("UTF-8")
 
 
+@overload
 def get_header_value(
     headers: dict[str, Any],
     name: str,
-    default_value: str | None,
-    case_sensitive: bool | None,
-) -> str | None:
```

Comment:
This should not work with Python < 3.10

Reply:
I believe the presence of "from __future__ import annotations" makes it work :)

Our CI runs on all versions of the supported Python runtimes, starting at 3.8

Reply:
I looked into some other files. There are already a lot of files with `from __future__ import annotations`. I decide to revert the `Optional` change.

Reply:
The doc-string say `Default is None`. Before this commit `case_sensitive` was unset.

My feeling is, it should be `False` by default and never `None`.

Reply:
Agree!

Reply:
I will keep this weird behaviour because there are a lot of implementations that have the same.

I will move this to a new PR.